server/controllers/request.js

```diff
@@ -0,0 +1,77 @@
+const Request = require("../Models/Request");
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+
+// @route GET /request
+// @desc List of requests for logged in user
+// @access Private
+exports.getRequests = asyncHandler(async (req, res, next) => {
+  const userId = req.user.id;
+
+  const requests = await Request.find({
+    $or: [{ userId: userId }, { sitterId: userId }],
+  });
+
+  res.send({ requests });
+});
+
+// @route POST /request
+// @desc Create a new request
+// @access Private
+exports.createRequest = asyncHandler(async (req, res, next) => {
+  const userId = req.user.id;
+
+  const { sitterId, startDate, endDate } = req.body;
+
+  if (!startDate || !endDate) {
+    res.status(400);
+    throw new Error("Incomplete date field");
+  }
+
+  const sitterExists = await User.findById(sitterId);
+
+  if (!sitterExists) {
+    res.status(400);
+    throw new Error("Invalid sitter id");
+  }
+
+  const request = {
+    userId: userId,
+    sitterId: sitterId,
+    startDate: startDate,
+    endDate: endDate,
+  };
+
+  const resp = await Request.create(request);
+
+  res.status(201).json(resp);
```

Comment:
Question: Where is the error handling performed? :)

Reply:
userId should be valid as the user must be logged it to the protected route, sitterExists has been checked. Mongoose will throw an error on save if an invalid date is entered. Is a more explicit error check for dates expected at this point?

Reply:
@simalicrum I was referring error handling in the event the document creation has failed? :)